google-cloud-firestore/src/main/java/com/google/cloud/firestore/FieldMask.java

```diff
@@ -47,7 +45,7 @@ private FieldMask(SortedSet<FieldPath> fieldPaths) {
    */
   @Nonnull
   public static FieldMask of(String... fieldPaths) {
-    List<FieldPath> paths = new ArrayList<>();
+    TreeSet<FieldPath> paths = new TreeSet<>();
```

Comment:
The change from ArrayList to TreeSet in this class looks like strictly _worse_ performance, since adding to a TreeSet is slower than adding to an ArrayList. Also, using TreeSet subtly changes the semantics because a TreeSet will eliminate duplicates and potentially change the order of elements where an ArrayList would keep them. Could you explain the motivation for changing to TreeSet?

Reply:
Nevermind. @tom-andersen pointed out to me that TreeSet was already there; this change avoids creating an intermediary collection that is (almost) immediately converted to a TreeSet then thrown away.